editor/editor_properties.cpp

```diff
@@ -2183,7 +2183,55 @@ void EditorPropertyResource::_menu_option(int p_which) {
 		case OBJ_MENU_NEW_SCRIPT: {
 
 			if (Object::cast_to<Node>(get_edited_object())) {
-				EditorNode::get_singleton()->get_scene_tree_dock()->open_script_dialog(Object::cast_to<Node>(get_edited_object()));
+				if (get_edited_property() == "script") {
+					EditorNode::get_singleton()->get_scene_tree_dock()->open_script_dialog(Object::cast_to<Node>(get_edited_object()));
+				} else {
+					// Copied from editor/scene_tree_dock.cpp#L380 and edited for our purposes
+					// There's a good chance we could merge this back into that code with some modifications
+					//Ref<Script> existing = get_edited_object()->get(get_edited_property());
+					//Script *existing = Object::cast_to<Script>(get_edited_object()->get(get_edited_property()));
+					RES existing = get_edited_object()->get(get_edited_property());
+
+					String inherits;
+					if (existing.is_valid()) {
+						//inherits = existing->get_class();
+						for (int i = 0; i < ScriptServer::get_language_count(); i++) {
+							ScriptLanguage *l = ScriptServer::get_language(i);
+							if (l->get_type() == existing->get_class()) {
+								String name = l->get_global_class_name(existing->get_path());
+								if (ScriptServer::is_global_class(name) && EDITOR_GET("interface/editors/derive_script_globals_by_name").operator bool()) {
+									inherits = name;
+								} else if (l->can_inherit_from_file()) {
+									inherits = "\"" + existing->get_path() + "\"";
+								}
+								break;
+							}
+						}
+					} else {
+						inherits = "Resource";
+					}
+
+					Node *edited_node = Object::cast_to<Node>(get_edited_object());
+
+					String path = edited_node->get_filename().get_basename();
+					if (path == "") {
+						String root_path = EditorNode::get_singleton()->get_editor_data().get_edited_scene_root()->get_filename();
+						if (root_path == "") {
+							path = String("res://").plus_file(edited_node->get_name());
+						} else {
+							path = root_path.get_base_dir().plus_file(edited_node->get_name());
+						}
+					}
+					// path = <node_or_scene_name>.<property_name>.
+					// The extra period at the end is to prevent the property name from being seen as the type by ScriptCreateDialog
+					path = path + "." + get_edited_property() + ".";
+
+					ScriptCreateDialog *script_create_dialog = EditorNode::get_singleton()->get_scene_tree_dock()->get_script_create_dialog();
+					script_create_dialog->connect("script_created", this, "_script_created");
+					script_create_dialog->connect("popup_hide", this, "_script_creation_closed");
```

Comment:
I realized a couple things shortly after making my other PR that changed the SceneTreeDock to only listen for script creation when it's the one that opened the dialog.

First off, I probably should have marked the `popup_hide` signal connection as CONNECT_ONESHOT. This would of saved a single line of code at most, though. Should I do that here (and maybe change it to CONNECT_ONESHOT in SceneTreeDock's code too)?

Second, I initially brushed off the idea of adding the property name as a signal bind, but now I'm wondering if maybe I should have done that to prevent this extra code.

Thoughts?

Reply:
> First off, I probably should have marked the popup_hide signal connection as CONNECT_ONESHOT. This would of saved a single line of code at most, though. Should I do that here (and maybe change it to CONNECT_ONESHOT in SceneTreeDock's code too)?

Sounds good.

> Second, and this is related to the review comment I made about merging this into scene_tree_dock.cpp, I initially brushed off the idea of adding the property name that the script should be attached to, as a signal bind, but now I'm wondering if maybe I should have done that to try to prevent this extra code from needing to be used.

I'm not familiar enough with this code to say, but if it makes the code simpler/cleaner, that would be great.

Reply:
I'm not sure how clean I'll be able to make it, so... I will merge it in its own commit, and then we can decide if the way it's been handled is better or not.